src/global/shared-types/link.types.ts

```diff
@@ -0,0 +1,28 @@
+export interface Link {
+    /**
+     * The url the link should point to.
+     */
+    href: string;
+
+    /**
+     * The text value to use for the link.
+     * Note that this might not be used by all components that use the
+     * Link interface.
+     */
+    text?: string;
+
+    /**
+     * Title for the link. Read by assistive tech and shown when the
+     * link is hovered. Can be used to provide additional information
+     * about the link. It improves accessibility both for sighted users
+     * and users of assistive technologies.
+     */
+    title?: string;
+
+    /**
+     * Target for the link. See
+     * https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-target
+     * for more info.
+     */
+    target?: string;
```

Comment:
just curious. why didn't we do this: 
```tsx
target?: '_self' | '_blank' | '_parent' | '_top' = '_self';
```

Reply:
Because those are not the only values allowed. Any string is allowed, those are only the values that have special meaning. It was ages since I used anything but the default or `_blank`, so I'm not sure, but if I recall correctly, setting any other value will cause the link to open a new window or tab, and set the name of that "context" to the value you entered. If you then click another link on the first page, which uses the same target value, that link will open in the window or tab that was opened when you clicked the first link.